src/ml_tooling/baseclass.py

```diff
@@ -377,21 +384,35 @@ def gridsearch(self,
             ResultGroup object containing each individual score
         """
 
+        baseline_model = clone(self.model)
+        train_x, train_y = indexable(self.data.train_x, self.data.train_y)
         metric = self.default_metric if metric is None else metric
+        n_jobs = self.config.N_JOBS if n_jobs is None else n_jobs
         cv = self.config.CROSS_VALIDATION if cv is None else cv
+        cv = check_cv(cv, train_y, baseline_model._estimator_type == 'classifier')  # Stratify?
```

Comment:
Nice find 👍 This lets us use any sklearn CV class if we need to

It will use a stratified KFold if y is a binary or multiclass, so that's a nice benefit too
https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.check_cv.html

Reply:
It can be used for time series too https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.TimeSeriesSplit.html#sklearn.model_selection.TimeSeriesSplit which is pretty neat.